runtime/cudaq/platform/default/rest_server/helpers/JsonConvert.h

```diff
@@ -0,0 +1,201 @@
+  }
+}
+
+// `ExecutionContext` serialization.
+inline void to_json(json &j, const ExecutionContext &context) {
+  j = json{{"name", context.name},
+           {"shots", context.shots},
+           {"hasConditionalsOnMeasureResults",
+            context.hasConditionalsOnMeasureResults}};
+
+  const auto &regNames = context.result.register_names();
+  // Here, we serialize the full lists of ExecutionResult records so that
+  // expectation values are captured.
+  std::vector<ExecutionResult> results;
+  for (const auto &regName : regNames) {
+    ExecutionResult result;
+    result.registerName = regName;
+    result.counts = context.result.to_map(regName);
+    result.sequentialData = context.result.sequential_data(regName);
+    if (context.result.has_expectation(regName))
+      result.expectationValue = context.result.expectation(regName);
+    results.emplace_back(std::move(result));
+  }
+  j["result"] = results;
+
+  if (context.expectationValue.has_value()) {
+    j["expectationValue"] = context.expectationValue.value();
+  }
+  j["simulationData"] = json();
+  j["simulationData"]["dim"] = std::get<0>(context.simulationData);
+  j["simulationData"]["data"] = std::get<1>(context.simulationData);
+  if (context.spin.has_value() && context.spin.value() != nullptr) {
+    const std::vector<double> spinOpRepr =
+        context.spin.value()->getDataRepresentation();
+    const auto spinOpN = context.spin.value()->num_qubits();
+    j["spin"] = json();
+    j["spin"]["num_qubits"] = spinOpN;
+    j["spin"]["data"] = spinOpRepr;
+  }
+  j["registerNames"] = context.registerNames;
+}
+
+inline void from_json(const json &j, ExecutionContext &context) {
+  j.at("shots").get_to(context.shots);
+  j.at("hasConditionalsOnMeasureResults")
+      .get_to(context.hasConditionalsOnMeasureResults);
+
+  if (j.contains("result")) {
+    std::vector<ExecutionResult> results;
+    j.at("result").get_to(results);
+    context.result = sample_result(results);
+  }
+
+  if (j.contains("expectationValue")) {
+    double expectationValue;
+    j["expectationValue"].get_to(expectationValue);
+    context.expectationValue = expectationValue;
+  }
+
+  if (j.contains("spin")) {
+    std::vector<double> spinData;
+    j["spin"]["data"].get_to(spinData);
+    const std::size_t nQubits = j["spin"]["num_qubits"];
+    auto serializedSpinOps = std::make_unique<spin_op>(spinData, nQubits);
+    context.spin = serializedSpinOps.release();
+  }
+
+  if (j.contains("simulationData")) {
+    std::vector<std::size_t> stateDim;
+    std::vector<std::complex<double>> stateData;
+    j["simulationData"]["dim"].get_to(stateDim);
+    j["simulationData"]["data"].get_to(stateData);
+    context.simulationData =
+        std::make_tuple(std::move(stateDim), std::move(stateData));
+  }
+
+  if (j.contains("registerNames"))
+    j["registerNames"].get_to(context.registerNames);
+}
+
+// Enum data to denote the payload format.
+enum CodeFormat { MLIR, LLVM };
```

Comment:
Can this be done with scoped enumeration?
https://www.geeksforgeeks.org/enum-classes-in-c-and-their-advantage-over-enum-datatype/

Reply:
Definitely. Fixed in https://github.com/NVIDIA/cuda-quantum/pull/1012/commits/67ad36aefb691b661293fdd46a5822b543e2243d